docs/docs_source/hostpage.rst

```diff
@@ -0,0 +1,97 @@
+benefits, including:
+
+#. Since the Office Online application is hosted within a host page, the host page can communicate with the frame
+   easily using :ref:`PostMessage <PostMessage>`. This allows a richer integration between the host and Office Online.
+#. URLs displayed in the user's browser are host URLs. This means that from a user's perspective, they are
+   interacting with the host, not Office Online. This also ensures that URLs that are copied and pasted by users are
+   host URLs, not Office Online URLs.
+
+The host page is typically very simple; it must meet only the following requirements:
+
+* It must use a ``form`` element and POST the :term:`access token` and access_token_ttl values to the Office Online
+  iframe :ref:`for security purposes <Passing access tokens securely>`.
+* It must include any JavaScript needed to interact with the Office Online iframe using
+  :ref:`PostMessage <PostMessage>`.
+* It must manage :ref:`wd* query string parameters <wd Parameters>`.
+* It must apply some specific CSS styles to the ``body`` element and Office Online iframe to avoid visual bugs.
+* It must declare a ``viewport`` meta tag to avoid visual and functional problems in mobile browsers.
+
+Host page example
+-----------------
+
+The `Office Online GitHub repository <repo>`_ contains a `minimal example host page <SampleHostPage.html>`_. Note
+that it does not illustrate managing :ref:`wd* query string parameters <wd Parameters>` or :ref:`PostMessage`. The
+sections below will refer to it in more detail.
+
+
+..  _Passing access tokens securely:
+
+Passing access tokens securely
+------------------------------
+
+It is important, for security purposes, that access tokens not be passed to the Office Online iframe as a query
+string parameter. Doing so would greatly increase the likelyhood of token leakage. To avoid this problem, hosts
+should pass the access token and access_token_ttl values to the Office Online iframe using a form POST. This technique
+is illustrated in the sample host page:
+
+..  literalinclude:: ../../samples/SampleHostPage.html
+    :caption: Sample code from `SampleHostPage.html`_ illustrating how to submit access tokens securely
+    :language: html
+    :linenos:
+    :lineno-match:
+    :lines: 30-39
+
+In an actual implementation, the ``<OFFICE_ONLINE_ACTION_URL>``, ``<ACCESS_TOKEN_VALUE>``, and
+``<ACCESS_TOKEN_TTL_VALUE>`` strings should be replaced with appropriate values.
+
+
+..  _wd Parameters:
+
+Passing through wd* parameters
+------------------------------
+
+Office Online will sometimes pass additional query string parameters to your host page. These query string parameters
+are of the form ``wd*``. When you receive these query string parameters on your host page URLs, you must pass them,
+unchanged, to the Office Online iframe.
+
+In addition, if the `replaceState`_ method from the HTML5 History API is available in the user's browser, you should
```

Comment:
That's an [external link](http://sphinx-doc.org/rest.html#external-links) in reStructuredText. :) The link target is at the bottom of the file on line 96.